File: sdk/dart/dovahlink_client/lib/src/protocol/pairing_ack_payload.dart
import 'package:json_annotation/json_annotation.dart';

import 'json_map.dart';

part 'pairing_ack_payload.g.dart';

/// Outgoing `pairing_ack` payload (`protocol/schema/README.md`'s `pairing_ack`). Encode-only: the
/// client never decodes its own `pairing_ack`.
@JsonSerializable(createFactory: false)
class PairingAckPayload {
  /// Creates a pairing-ack payload.
  const PairingAckPayload({required this.credential});

  /// The hex-encoded credential received in a prior `credential_issued` outcome.
  final String credential;

  /// Encodes this payload as a JSON object.
  JsonMap toJson() => _$PairingAckPayloadToJson(this);
}

File: integration/DovahLinkValidationClient/PairingAckPayload.cs
using System.Text.Json.Nodes;

namespace DovahLinkValidationClient;

/// <summary>
/// Outgoing <c>pairing_ack</c> payload (<c>protocol/schema/README.md</c>'s <c>pairing_ack</c>).
/// Encode-only: this client never decodes its own <c>pairing_ack</c>.
/// </summary>
/// <param name="Credential">The hex-encoded credential received in a prior
/// <c>credential_issued</c> outcome, saved to persistent storage before this message is sent.</param>
public sealed record PairingAckPayload(string Credential)
{
    /// <summary>
    /// Encodes this payload as a JSON object.
    /// </summary>
    /// <returns>The encoded <c>pairing_ack</c> payload.</returns>
    public JsonObject Encode() => new() { ["credential"] = Credential };
}
